pandas/core/common.py

```diff
@@ -2510,6 +2510,8 @@ def is_floating_dtype(arr_or_dtype):
 
 
 def is_bool_dtype(arr_or_dtype):
+    if arr_or_dtype is NotImplemented:
```

Comment:
sounds good, one last thing. I tested the following and it also crapped out.

```
_get_dtype_type(False)
```

so is this also supposed to crap out? or is it supposed to return False?

```
is_bool_dtype(False)
```